omegatrade-app/backend/app/controller/user.controller.js

```diff
@@ -73,72 +45,93 @@ exports.register = async function (req, res) {
  * @method POST
  * 
- * @param {*} req contains request headers and payload businessEmail,password
+ * @param {*} req contains request headers and payload businessEmail,password,provider,photoUrl
  */
-exports.getToken = function (req, res) {
-    const body = req.body;
-    if (body && body.provider === 'GOOGLE') {
-        User.findUser(body.email, function (err, user) {
-            if (user && user.userId) {
-                    delete user.password;
-                    const token = jwt.sign(user, process.env.JWT_SECRET, {
-                        expiresIn: process.env.EXPIRE_IN
-                    });
-                    res.status(200).json({
-                        success: true,
-                        message: 'Logged in successfully',
-                        userInfo: user,
-                        authToken: token
-                    });
-                
-            } else
-                res.status(401).json({
-                    success: false,
-                    message: 'please register your account',
-                    redirect: 'sign-up'
-                });
-            if (err)
-                res.status(500).json({
-                    success: false,
-                    message: 'Something went wrong,Error while authenticating user!'
-                });
-        });
+exports.googleSignIn = async function (req, res) {
+    try {
+        const body = req.body;
+        const [user] = await User.findUser(body.email)
+        if (user && user.userId) {
+            // updating provider and photoUrl only if user did not registered with goolge before
+            if (user.provider === '' || user.provider === null) {
+                user.provider = body.provider;
+                user.photoUrl = body.photoUrl;
+                await User.update(user)
+            } 
+            const token = jwt.sign(user, process.env.JWT_KEY, {
+                expiresIn: process.env.EXPIRE_IN
+            });
+            return res.status(200).json({ success: true, message: 'Logged in successfully', userInfo: user, authToken: token });
+        } else {
+            const salt = await bcrypt.genSalt(10);
+            // generating random password as password field should not empty.
+            const randomPassword = Math.random().toString(36).slice(-6);
+            const password = await bcrypt.hash(randomPassword, salt);
+            const user = {
+                userId: uuidv4(),
+                businessEmail: body.email,
+                fullName: body.name,
+                password: password,
+                photoUrl: body.photoUrl,
+                provider: body.provider,
+            };
+            await User.registerUser(user);
+            jwt.sign(user, process.env.JWT_KEY, { expiresIn: process.env.EXPIRE_IN }, function (err, token) {
+                if (err) {
+                    return res.status(409).json({ success: false, message: 'Something went wrong while registering new user!' });
```

Comment:
nit: error code `409` seems out of place here. `409` indicates a conflict, I don't think that is correct here.